modules/minimum-required-permissions-upi-azure.adoc

```diff
@@ -0,0 +1,239 @@
+* `Microsoft.Network/privateDnsZones/write`
+* `Microsoft.Network/publicIPAddresses/delete`
+* `Microsoft.Network/publicIPAddresses/join/action`
+* `Microsoft.Network/publicIPAddresses/read`
+* `Microsoft.Network/publicIPAddresses/write`
+* `Microsoft.Network/virtualNetworks/join/action`
+* `Microsoft.Network/virtualNetworks/read`
+* `Microsoft.Network/virtualNetworks/subnets/join/action`
+* `Microsoft.Network/virtualNetworks/subnets/read`
+* `Microsoft.Network/virtualNetworks/subnets/write`
+* `Microsoft.Network/virtualNetworks/write`
+====
+
+.Required permissions for checking health of resources
+[%collapsible]
+====
+* `Microsoft.Resourcehealth/healthevent/Activated/action`
+* `Microsoft.Resourcehealth/healthevent/InProgress/action`
+* `Microsoft.Resourcehealth/healthevent/Pending/action`
+* `Microsoft.Resourcehealth/healthevent/Resolved/action`
+* `Microsoft.Resourcehealth/healthevent/Updated/action`
+====
+
+.Required permissions for creating resource groups
+[%collapsible]
+====
+* `Microsoft.Resources/subscriptions/resourceGroups/read`
+* `Microsoft.Resources/subscriptions/resourcegroups/write`
+====
+
+.Required permissions for creating resource tags
+[%collapsible]
+====
+* `Microsoft.Resources/tags/write`
+====
+
+.Required permissions for creating storage resources
+[%collapsible]
+====
+* `Microsoft.Storage/storageAccounts/blobServices/read`
+* `Microsoft.Storage/storageAccounts/blobServices/containers/write`
+* `Microsoft.Storage/storageAccounts/fileServices/read`
+* `Microsoft.Storage/storageAccounts/fileServices/shares/read`
+* `Microsoft.Storage/storageAccounts/fileServices/shares/write`
+* `Microsoft.Storage/storageAccounts/fileServices/shares/delete`
+* `Microsoft.Storage/storageAccounts/listKeys/action`
+* `Microsoft.Storage/storageAccounts/read`
+* `Microsoft.Storage/storageAccounts/write`
+====
+
+.Required permissions for creating deployment
+[%collapsible]
+====
+* `Microsoft.Resources/deployments/read`
+* `Microsoft.Resources/deployments/write`
+* `Microsoft.Resources/deployments/validate/action`
+* `Microsoft.Resources/deployments/operationstatuses/read`
+====
+
+.Optional permissions for creating marketplace virtual machine resources
+[%collapsible]
+====
+* `Microsoft.MarketplaceOrdering/offertypes/publishers/offers/plans/agreements/read`
+* `Microsoft.MarketplaceOrdering/offertypes/publishers/offers/plans/agreements/write`
+====
+
+.Optional permissions for creating compute resources
+[%collapsible]
+====
+* `Microsoft.Compute/diskEncryptionSets/read`
+* `Microsoft.Compute/diskEncryptionSets/write`
+* `Microsoft.Compute/diskEncryptionSets/delete`
+====
+
+.Optional resources for creating key vault resources
+[%collapsible]
+====
+* `Microsoft.KeyVault/vaults/read`
+* `Microsoft.KeyVault/vaults/write`
+* `Microsoft.KeyVault/vaults/delete`
+* `Microsoft.KeyVault/vaults/deploy/action`
+* `Microsoft.KeyVault/vaults/keys/read`
+* `Microsoft.KeyVault/vaults/keys/write`
+====
```

Comment:
Please merge them as done in IPI part with : "Optional permissions for enabling user-defined disk encryption set"

And as discussed in [CORS-2559](https://issues.redhat.com/browse/CORS-2559), please add `Microsoft.Features/providers/features/register/action` here.

Reply:
Addressed